core/alea/aba/aba.go

```diff
@@ -0,0 +1,385 @@
+package aba
+
+import (
+	"context"
+
+	"github.com/obolnetwork/charon/app/log"
+	"github.com/obolnetwork/charon/app/z"
+	"github.com/obolnetwork/charon/tbls"
+)
+
+// Asynchronous Binary Agreement implementation from the paper: "Cobalt: BFT Governance in Open Networks"
+// Link: https://arxiv.org/pdf/1802.07240.pdf (Page 19/20)
+
+type MsgType uint
+
+const (
+	MsgUnknown MsgType = iota
+	MsgInit
+	MsgAux
+	MsgConf
+	MsgFinish
```

Comment:
Add comments here what each message means or maybe use more meaningful names of these constants.